rllib/agents/trainer.py

```diff
@@ -1048,15 +1050,24 @@ def make_callbacks():
 
     @staticmethod
     def _validate_config(config: PartialTrainerConfigDict):
-        if config.get("_use_trajectory_view_api") and \
-                config.get("framework") != "torch":
+        use_traj_view_api = config.get("_use_trajectory_view_api")
+        model_is_auto_lstm = config.get("model", {}).get("use_lstm")
+        # Enable trajectory view API iff None and framework=torch and
+        # model=auto-LSTM (meaning it's a time_major model).
+        if use_traj_view_api is None:
+            if config.get("framework") == "torch" and model_is_auto_lstm:
+                config["_use_trajectory_view_api"] = True
+            else:
+                config["_use_trajectory_view_api"] = False
+        elif use_traj_view_api and config.get("framework") != "torch":
             raise ValueError(
                 "`_use_trajectory_view_api` only supported for PyTorch so "
                 "far!")
-        elif not config.get("_use_trajectory_view_api") and \
-                config.get("model", {}).get("_time_major"):
-            raise ValueError("`model._time_major` only supported "
-                             "iff `_use_trajectory_view_api` is True!")
+
+        # Switch on time-major, iff trajectory view API and model is torch
+        # and auto-lstm.
+        if config["_use_trajectory_view_api"] and model_is_auto_lstm:
+            config["model"]["_time_major"] = True
```

Comment:
I think we should only flip to to False programmatically. For setting it to True, that should come from the agent config. That way it's pretty clear: it's on by default for certain algos, except when it's not supported (e.g., TensorFlow).